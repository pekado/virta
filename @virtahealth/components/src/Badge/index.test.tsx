import * as React from "react";
import { createComponentWithVirtaContext } from "../test";
import { Badge } from "./index";

const defaultProps = {
  size: 25,
  label: "1",
};

describe("Badge Component", () => {
  it("Renders a badge with default color", () => {
    const wrapper = createComponentWithVirtaContext(
      <Badge {...defaultProps} />
    );
    expect(wrapper.getByText("1")).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders a badge with a 0 label", () => {
    const wrapper = createComponentWithVirtaContext(
      <Badge {...defaultProps} label="0" />
    );
    expect(wrapper.getByText("0")).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders an orange badge", () => {
    const wrapper = createComponentWithVirtaContext(
      <Badge {...defaultProps} color="orange" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders a blue badge", () => {
    const wrapper = createComponentWithVirtaContext(
      <Badge {...defaultProps} color="blue" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders a pink badge", () => {
    const wrapper = createComponentWithVirtaContext(
      <Badge {...defaultProps} color="pink" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders a badge using HTML color", () => {
    const wrapper = createComponentWithVirtaContext(
      <Badge {...defaultProps} color="yellow" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders a badge using RGB", () => {
    const wrapper = createComponentWithVirtaContext(
      <Badge {...defaultProps} color="rgb(0, 0, 0)" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders an empty badge", () => {
    const wrapper = createComponentWithVirtaContext(
      <Badge {...defaultProps} label={undefined} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
