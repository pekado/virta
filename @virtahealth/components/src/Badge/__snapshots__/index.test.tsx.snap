// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component Renders a badge using HTML color 1`] = `
<View
  color="yellow"
  size={25}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "yellow",
        "borderRadius": 25,
        "height": 25,
        "justifyContent": "center",
        "width": 25,
      },
    ]
  }
>
  <Text
    size={25}
    style={
      Array [
        Object {
          "color": "rgb(36, 38, 43)",
          "fontFamily": "Whitney-Book",
          "fontSize": 20,
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 20,
          "textDecorationColor": "black",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
        },
        Object {
          "color": "rgb(255,255,255)",
          "fontFamily": "Whitney-Medium",
          "fontSize": 16.666666666666668,
          "fontWeight": "700",
          "textAlign": "center",
          "textTransform": "uppercase",
          "width": "100%",
        },
      ]
    }
  >
    1
  </Text>
</View>
`;

exports[`Badge Component Renders a badge using RGB 1`] = `
<View
  color="rgb(0, 0, 0)"
  size={25}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(0, 0, 0)",
        "borderRadius": 25,
        "height": 25,
        "justifyContent": "center",
        "width": 25,
      },
    ]
  }
>
  <Text
    size={25}
    style={
      Array [
        Object {
          "color": "rgb(36, 38, 43)",
          "fontFamily": "Whitney-Book",
          "fontSize": 20,
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 20,
          "textDecorationColor": "black",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
        },
        Object {
          "color": "rgb(255,255,255)",
          "fontFamily": "Whitney-Medium",
          "fontSize": 16.666666666666668,
          "fontWeight": "700",
          "textAlign": "center",
          "textTransform": "uppercase",
          "width": "100%",
        },
      ]
    }
  >
    1
  </Text>
</View>
`;

exports[`Badge Component Renders a badge with a 0 label 1`] = `
<View
  size={25}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(22, 56, 76)",
        "borderRadius": 25,
        "height": 25,
        "justifyContent": "center",
        "width": 25,
      },
    ]
  }
>
  <Text
    size={25}
    style={
      Array [
        Object {
          "color": "rgb(36, 38, 43)",
          "fontFamily": "Whitney-Book",
          "fontSize": 20,
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 20,
          "textDecorationColor": "black",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
        },
        Object {
          "color": "rgb(255,255,255)",
          "fontFamily": "Whitney-Medium",
          "fontSize": 16.666666666666668,
          "fontWeight": "700",
          "textAlign": "center",
          "textTransform": "uppercase",
          "width": "100%",
        },
      ]
    }
  >
    0
  </Text>
</View>
`;

exports[`Badge Component Renders a badge with default color 1`] = `
<View
  size={25}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(22, 56, 76)",
        "borderRadius": 25,
        "height": 25,
        "justifyContent": "center",
        "width": 25,
      },
    ]
  }
>
  <Text
    size={25}
    style={
      Array [
        Object {
          "color": "rgb(36, 38, 43)",
          "fontFamily": "Whitney-Book",
          "fontSize": 20,
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 20,
          "textDecorationColor": "black",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
        },
        Object {
          "color": "rgb(255,255,255)",
          "fontFamily": "Whitney-Medium",
          "fontSize": 16.666666666666668,
          "fontWeight": "700",
          "textAlign": "center",
          "textTransform": "uppercase",
          "width": "100%",
        },
      ]
    }
  >
    1
  </Text>
</View>
`;

exports[`Badge Component Renders a blue badge 1`] = `
<View
  color="blue"
  size={25}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(22, 56, 76)",
        "borderRadius": 25,
        "height": 25,
        "justifyContent": "center",
        "width": 25,
      },
    ]
  }
>
  <Text
    size={25}
    style={
      Array [
        Object {
          "color": "rgb(36, 38, 43)",
          "fontFamily": "Whitney-Book",
          "fontSize": 20,
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 20,
          "textDecorationColor": "black",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
        },
        Object {
          "color": "rgb(255,255,255)",
          "fontFamily": "Whitney-Medium",
          "fontSize": 16.666666666666668,
          "fontWeight": "700",
          "textAlign": "center",
          "textTransform": "uppercase",
          "width": "100%",
        },
      ]
    }
  >
    1
  </Text>
</View>
`;

exports[`Badge Component Renders a pink badge 1`] = `
<View
  color="pink"
  size={25}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(203, 0, 113)",
        "borderRadius": 25,
        "height": 25,
        "justifyContent": "center",
        "width": 25,
      },
    ]
  }
>
  <Text
    size={25}
    style={
      Array [
        Object {
          "color": "rgb(36, 38, 43)",
          "fontFamily": "Whitney-Book",
          "fontSize": 20,
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 20,
          "textDecorationColor": "black",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
        },
        Object {
          "color": "rgb(255,255,255)",
          "fontFamily": "Whitney-Medium",
          "fontSize": 16.666666666666668,
          "fontWeight": "700",
          "textAlign": "center",
          "textTransform": "uppercase",
          "width": "100%",
        },
      ]
    }
  >
    1
  </Text>
</View>
`;

exports[`Badge Component Renders an empty badge 1`] = `
<View
  size={25}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(22, 56, 76)",
        "borderRadius": 25,
        "height": 25,
        "justifyContent": "center",
        "width": 25,
      },
    ]
  }
/>
`;

exports[`Badge Component Renders an orange badge 1`] = `
<View
  color="orange"
  size={25}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(127, 58, 5)",
        "borderRadius": 25,
        "height": 25,
        "justifyContent": "center",
        "width": 25,
      },
    ]
  }
>
  <Text
    size={25}
    style={
      Array [
        Object {
          "color": "rgb(36, 38, 43)",
          "fontFamily": "Whitney-Book",
          "fontSize": 20,
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 20,
          "textDecorationColor": "black",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
        },
        Object {
          "color": "rgb(255,255,255)",
          "fontFamily": "Whitney-Medium",
          "fontSize": 16.666666666666668,
          "fontWeight": "700",
          "textAlign": "center",
          "textTransform": "uppercase",
          "width": "100%",
        },
      ]
    }
  >
    1
  </Text>
</View>
`;
